js/src/Axis.ts

```diff
@@ -119,11 +121,13 @@ export class Axis extends WidgetView {
                 const numDateTicks = (this.width < DATESCALE_WIDTH_THRESHOLD) ?
                                         5 :
                                         undefined;
-                this.axis.tickValues(this.axis_scale.scale.ticks(numDateTicks));
+                const scale = this.axis_scale.scale as d3.ScaleTime<Date, number>;
```

Comment:
For scales, we won't need these `as ...`. This is handle by this PR https://github.com/bqplot/bqplot/pull/904